### PR TITLE
Localization cleanup for 3.0

### PR DESCRIPTION
**Pull request by matthiasgeiger**
- [x] Adds missing keys to JabRef_en.properties
- [x] Check whether alternative, already existing keys can be reused
- [x] remove obsolete keys
- [x] Cleanup discrepancies listed by mlep
- [x] Add missing keys for all other languages
- [x] Integrate search localizations
- [ ] Resort? Non-Empty Alphebetically followed by Emphy Alphabetically?

### CODE REVIEW

File: src/main/java/net/sf/jabref/JabRef.java [hunk 1]
```diff
@@ -352,7 +352,7 @@ private void setupLogHandlerForErrorConsole() {
                                 }
                                 session.commit();
                             } catch (SaveException ex) {
-                                System.err.println(Localization.lang("Could not save file") + " '" + data[0] + "': "
+                                System.err.println(Localization.lang("Could not save file.") + "\n"
```
Comment (simonharrer) [inline src/main/java/net/sf/jabref/JabRef.java]: Why not use Could not save file %0. to print out the file name as well?

**Comment by simonharrer** [comment]
:+1:

**Comment by koppor** [comment]
@kojiyokota Also added added some missing keys to [JabRef_ja.properties](https://github.com/JabRef/jabref/blob/1332070cb62a8990567831a10f026d31acfc1738/src/main/resources/l10n/JabRef_ja.properties). Don't know if that information comes too late.

**Comment by matthiasgeiger** [comment]
@koppor Not a problem - I just wondered why some keys are already present in the Japanese l10n :wink:

**Comment by mlep** [comment]
I have looked at the discrepancies between the keys and their English localization.
Since it is clean-up time, some keys should be updated. And since it is better for everybody that I do not modify the java code :wink:, here is the list.  

Based on the master branch,
In  **Menu_en.properties**:
```
Find_And_Fix=     should be Find_and_Fix=
Save_selected_as_plain_BibTeX_...= should be Save_selected_as_plain_BibTeX...=
```

In  **JabRef_en.properties**:
```
Add_Group= should be Add_group=
Add_Subgroup= should be Add_subgroup=
An_Exception_ocurred_while_accessing_'%0'= should be An_Exception_occurred_while_accessing_'%0'=
An_SAXException_ocurred_while_parsing_'%0'\:= should be An_SAXException_occurred_while_parsing_'%0'\:=
Available_fields= should be Allowed_BibTeX_fields=
                                                    (based on the context of use)
Bibtex_entry_creation= should be BibTeX_entry_creation=
Cite_selected_entries= should be Cite_selected_entries_between_parenthesis=
                                                    (based on the context of use)
Cut_pr= should be Cut=
Database_\:= should be Database\:=
Edit_file_link= should be Save_file=
                                                    (based on the context of use)
Error_occured_when_parsing_entry= should be Error_occurred_when_parsing_entry=
Import_Metadata_from:= should be Import_metadata_from\:=
Input= should be= Work options= 
                                                    (based on the context of use)
Messages_and_Hints= should be Hints_and_Warnings=
                                                    (based on the context of use)
no_base-bibtex-file_specified= should be no_base-BibTeX-file_specified=
No_exceptions_have_ocurred.= should be No_exceptions_have_occurred.=
Password_:= should be Password:=
Remove_all_subgroups= should be Remove_subgroups=
                                                    (based on the context of use)
Reset_file_type_definitons= should be Reset_file_type_definitions=
Server_Hostname_:= should be Server_Hostname\:=
Server_Type_:= should be Server_Type\:=
Show_optional_fields= should be Show_optional_field=
Sort_Automatically= should be Sort_automatically=
Sort_fields_in_alphabeta_order_(as_ver_2.10)= should be Sort_fields_in_alphabetic_order_(as_ver_2.10)=
Sort_fields_in_old_fasion_(as_ver_2.9.2)= should be Sort_fields_in_old_fashion_(as_ver_2.9.2)=
Starts_the_import_of_bibtex_entries.= should be Starts_the_import_of_BibTeX_entries.=
Text_Input_Area= should be paste_text_here=
To_set_up,_go_to_<B>Tools_->_Manage_journal_abbreviations</B>= should be To_set_up,_go_to_<B>Options_->_Manage_journal_abbreviations</B>=
Unable_to_connect_to_freecite_online_service.= should be Unable_to_connect_to_FreeCite_online_service.=
Username_:= should be Username:=
```

**Comment by matthiasgeiger** [comment]
Thanks for another few hours work :wink:

**Comment by mlep** [comment]
@koppor should be blamed: he is the one pointing me to this thread... :wink:

**Comment by matthiasgeiger** [comment]
Phewww... should be ready to merge...

**Comment by matthiasgeiger** [comment]
Ahh one thing to be discussed... maybe @mlep can help here:

Should we resort the files somehow? 

All already translated keys to top - all untranslated to the end? 

Sorting translated/untranslated alphabetically?

**Comment by simonharrer** [comment]
I vote for alphabetically in total. 

Why:
- scripts can easily add entries
- no manual moving of untranslated into the translated section when translating
- version control can show nice diffs for adding translation texts

Btw. nice work here!

**Comment by matthiasgeiger** [comment]
In general I agree, but depending on the workflow of the translaters it might be useful to seperate all untranslated items from the ones already translated...

**Comment by simonharrer** [comment]
I would propose to merge this quickly, as it affects a lot of different lines in the code. And the effort should not be wasted.

**Comment by matthiasgeiger** [comment]
:+1: 

I'll add the remaining German translations directly in `master`.

**Comment by mlep** [comment]
To make things easier, when empty keys are added to a file, they are added at the end. This way the translators find them easily (not everybody is using Popeye). 
And, from time to time (typically near the release of a new version), we sort them alphabetically.

**Comment by koppor** [comment]
:+1: 

I would propose to group the entries by usage in the UI, so that related entries are appearing next to each other. I know that this is more difficult to maintain, but more easy for the translators.

I think [popeye](https://github.com/JabRef/popeye) can sort the columns according to empty entries.

**Comment by matthiasgeiger** [comment]
Mhmm... I will not group those ~1500 lines by UI usage :wink:

**Comment by ambro2** [comment]
Some comments about sorting the keys.
Sorting alphabetically could be ok, provided that untraslated keys are grouped at the end of the file. Using popeye could be helpful in locating untraslated keys, but once translated it's difficult to came back to the just translated key, maybe to reword or reuse part of the translation. Furthermore, in some cases I put comments above or below the translated keys (e.g. when the context is not clear): when sorting alphabetically the comments will be lost. BTW, could it be possible to put comments at the end of the line?
So, I second the proposal from @mlep: sorting alphabetically but keep untraslated keys at the end of the file.
Even better would be grouping by usage, but I understand the difficulty to mantain.

**Pull request merged**